Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,856 @@
+            var geometryInstancesLength = geometryInstances.length;
+            var attributes;
+
+            // If using PolylineColorAppearance, check if each instance has a color attribute.
+            if (this.appearance instanceof PolylineColorAppearance) {
+                for (i = 0; i < geometryInstancesLength; ++i) {
+                    attributes = geometryInstances[i].attributes;
+                    if (!defined(attributes) || !defined(attributes.color)) {
+                        throw new DeveloperError('All GeometryInstances must have color attributes to use PolylineColorAppearance with GroundPolylinePrimitive.');
+                    }
+                }
+            }
+
+            // Automatically create line width attributes
+            for (i = 0; i < geometryInstancesLength; ++i) {
+                var geometryInstance = geometryInstances[i];
+                attributes = geometryInstance.attributes;
+                if (!defined(attributes.width)) {
+                    attributes.width = new GeometryInstanceAttribute({
+                        componentDatatype : ComponentDatatype.UNSIGNED_BYTE,
+                        componentsPerAttribute : 1.0,
+                        value : [geometryInstance.geometry.width]
+                    });
+                }
+            }
+
+            primitiveOptions.geometryInstances = geometryInstances;
+            primitiveOptions.appearance = this.appearance;
+
+            primitiveOptions._createShaderProgramFunction = function(primitive, frameState, appearance) {
+                createShaderProgram(that, frameState, appearance);
+            };
+            primitiveOptions._createCommandsFunction = function(primitive, appearance, material, translucent, twoPasses, colorCommands, pickCommands) {
+                createCommands(that, appearance, material, translucent, colorCommands, pickCommands);
+            };
+            primitiveOptions._updateAndQueueCommandsFunction = function(primitive, frameState, colorCommands, pickCommands, modelMatrix, cull, debugShowBoundingVolume, twoPasses) {
+                updateAndQueueCommands(that, frameState, colorCommands, pickCommands, modelMatrix, cull, debugShowBoundingVolume);
+            };
+
+            this._primitive = new Primitive(primitiveOptions);
+            this._primitive.readyPromise.then(function(primitive) {
+                that._ready = true;
+
+                if (that.releaseGeometryInstances) {
+                    that.geometryInstances = undefined;
+                }
+
+                var error = primitive._error;
+                if (!defined(error)) {
+                    that._readyPromise.resolve(that);
+                } else {
+                    that._readyPromise.reject(error);
+                }
+            });
+        }
+        this._primitive.appearance = this.appearance;
```

Comment:
What is this was constructed with an appearance and changed to a `PolylineColorAppearance`, but all of the instances don't have a color attribute?

Reply:
At the moment it fails with a shader error, but I can wire in something a little more elegant.

`GroundPrimitive` might have the same problem now that it supports materials...